content/articles/sentiment-analysis-with-spacy-and-scikit-learn/index.md

```diff
@@ -0,0 +1,472 @@
+
+- "This app keeps on crashing. What should I do?": This shows "Need for assistance"
+
+For a practical guide on intent detection [click](https://rasa.com/blog/rasa-nlu-in-depth-part-1-intent-classification/)
+
+In this tutorial, we will build a `standard sentiment analysis`.
+
+### Dataset used
+
+We are working with three datasets: IMDb dataset, amazon dataset, and Yelp dataset. The datasets contain reviews of different products or services, the reviews are labeled either `1` to show a `positive` review or `0` for `negative` review.
+
+Our dataset is in form of text format which is to read and use by our model.
+
+A snip of the three datasets in text format is shown below.
+
+![Amazon dataset snip](/engineering-education/sentiment-analysis-with-spacy-and-scikit-learn/amazon-snip.jpg)
+
+To download the amazon dataset in a text format click [here](https://drive.google.com/file/d/1JYwslHzs_7bCxaS1XEDzMByn27-EAbez/view?usp=sharing)
+
+![Imdb dataset snip](/engineering-education/sentiment-analysis-with-spacy-and-scikit-learn/imdb-snip.jpg)
+
+To download the IMDb dataset in a text format click [here](https://drive.google.com/file/d/11_TT9aP-HvY0AmipuQOZV3JpXPCwS6d_/view?usp=sharing)
+
+![Yelp dataset snip](/engineering-education/sentiment-analysis-with-spacy-and-scikit-learn/yelp-snip.jpg)
+
+To download the yelp dataset in a text format click [here](https://drive.google.com/file/d/1eguxUNJWf9sMGqFqrJATLFWLIMxXuR0i/view?usp=sharing)
+
+After downloading all three datasets, let's load them into our machine.
+
+### Loading dataset
+
+To load our dataset we use `Pandas` packages which are used for data manipulation, analysis, and reading of our dataset.
+
+```python
+import pandas as pd
+```
+
+Let's use `Pandas` to read our three datasets as shown below.
+
+```python
+df_yelp = pd.read_table('yelp_labelled.txt')
+df_imdb = pd.read_table('imdb_labelled.txt')
+df_amz = pd.read_table('amazon_cells_labelled.txt')
+```
+
+Since we have three different datasets, we have to concatenate them together.
+
+```python
+frames = [df_yelp,df_imdb,df_amz]
+```
+
+### Adding headers
+
+Our dataset has no headers, we separate our dataset into two columns and give them heading names. It will have two columns, the first column which is named `Message` will contain the actual review text while the other column is named `Target` will contain the label of either `1` or `0`.
+
+```python
+for colname in frames:
+    colname.columns = ["Message","Target"]
+```
+
+Let's print the column names as shown below.
+
+```python
+for colname in frames:
+    print(colname.columns)
+```
+
+The output is as shown below.
+
+```bash
+for colname in frames:
+    print(colname.columns)
+Index(['Message', 'Target'], dtype='object')
+Index(['Message', 'Target'], dtype='object')
+Index(['Message', 'Target'], dtype='object')
+```
+
+We need to assign keys to our dataset so that we can know which each dataset belongs to. The dataset belongs into three groups `Yelp`, `IMDB`, and `Amazon`.
+
```

Comment:
There is a formatting error here. Python code snippets don't seem to be holding onto their syntax. Can you check if you have added an extra or missed out on a \` character?

Reply:
I have not found where there is a missing ` character, or an extra one.